base/utf16.jl

```diff
@@ -110,7 +110,9 @@ function is_valid_utf16(data::AbstractArray{UInt16})
     return i > n || !utf16_is_surrogate(data[i])
 end
 
-is_valid_utf16(s::UTF16String) = is_valid_utf16(s.data)
+isvalid(str::UTF16String) = isvalid(UTF16String, str.data)
+is_valid_utf16(str::UTF16String) = isvalid(UTF16String, str.data)
+is_valid_utf16(data::AbstractArray{UInt16}) = isvalid(UTF16String, data)
```

Comment:
Deprecations are done per-method, and the methods are defined by the `deprecate` macro.  That way when it comes time to remove them, we just need to delete the section in deprecated.jl and don't need to do any archaeology.  Take a look at what `macroexpand(:(Base.@deprecate f(x, y) g(2x, 2y)))` does as a simple example